development/tools/pkg/rendertemplates/genericcomponent.go

```diff
@@ -13,8 +13,8 @@ func (j *Job) changeExtraRefsBase(base string) {
 	}
 }
 
-// GenerateComponentJobs generates jobs for components
-func (r *RenderConfig) GenerateComponentJobs(global map[string]interface{}) {
+// AppendJobs generates jobs for components if necessary and appends pre/post/main jobs to the list if they exist
+func (r *RenderConfig) AppendJobs(global map[string]interface{}) {
```

Comment:
AppendJobs should rather accept generated jobs as argument and append them. Generating of jobs should be called outside this function. This function is doing much more than appending.